src/test/java/org/jenkinsci/plugins/envinject/EnvInjectPluginActionTest.java

```diff
@@ -0,0 +1,155 @@
+package org.jenkinsci.plugins.envinject;
+
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.Item;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.model.User;
+import hudson.security.ACL;
+import hudson.security.AccessControlled;
+import hudson.security.GlobalMatrixAuthorizationStrategy;
+import hudson.security.Permission;
+import javax.annotation.Nonnull;
+import jenkins.model.Jenkins;
+import org.acegisecurity.Authentication;
+import org.acegisecurity.context.SecurityContext;
+import org.acegisecurity.context.SecurityContextHolder;
+import org.acegisecurity.userdetails.UsernameNotFoundException;
+import static org.junit.Assert.*;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.Before;
+import org.jvnet.hudson.test.CaptureEnvironmentBuilder;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.SingleFileSCM;
+
+/**
+ * Tests for {@link EnvInjectPluginAction}.
+ * @author Oleg Nenashev
+ */
+public class EnvInjectPluginActionTest {
+        
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+    
+    private FreeStyleProject p;
+    private CaptureEnvironmentBuilder capture;
+    
+    @Before
```

Comment:
I don't remember exactly the under what circumstances it failed (will file the issue next time I see it. Probably it was ```@BeforeClass```/```@AfterClass```). Nevermind, if it's working here, it's working.